### PR TITLE
Add stemmer and basic noun morphology functionality

### LINKED ISSUE #2049: Add noun morphology for German
We should be able to recognize nouns in their different morphological forms in German.

Parent #2095.

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry:

* [not-user-facing] Adds a German stemmer and functionality to create German noun forms.

## Relevant technical choices:

* The stemmer was adapted from an existing stemmer; authorship was attributed accordingly. The code was adapted from this specific [JavaScript implementation](http://snowball.tartarus.org/otherlangs/german_javascript.txt) of an [existing German stemming algorithm](http://snowball.tartarus.org/algorithms/german/stemmer.html).

## Test instructions

This PR can be tested by following these steps:

* Make sure you have the premium configuration set to the right branch (see https://github.com/Yoast/YoastSEO.js-premium-configuration/pull/7).
* Change the language of your site to German.
* Add a German text and a keyphrase (consisting of German nouns and function words; verbs and adjectives aren't supported yet).
* Make sure that all forms of the keyphrase that occur in the text are found. More extensive testing of various word forms is done by @christinatsaousi 

Fixes #2049

### CODE REVIEW

File: src/morphology/german/stem.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		}
+	}
+	return word;
+};
+
+/**
+ * Stems German words.
+ *
+ * Stem function adapted from:
+ * Author: Joder Illi
```
Comment (hansjovis) [inline src/morphology/german/stem.js]: Perhaps we can markup this JSDoc comment with some fancy tags like [`@author`](http://usejsdoc.org/tags-author.html) and [`@copyright`](http://usejsdoc.org/tags-copyright.html) 🙂.
Although I am not sure whether `@copyright` works in this context.

File: src/morphology/german/getForms.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	return forms;
+};
+
+/**
+ * Creates morphological forms for a given German word.
+ *
+ * @param {string} word             The word to create the forms for.
+ * @param {Object} morphologyData   The German morphology data (false if unavailable).
+ *
+ * @returns {Array<string>} Array of the created
```
Comment (hansjovis) [inline src/morphology/german/getForms.js]: Description has not been
😃.

File: src/morphology/german/getForms.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+ */
+export function getForms( word, morphologyData ) {
+	const stemmedWord = stem( word );
+	const forms = new Array( word );
+	const exceptions = checkExceptions( morphologyData.nouns, stemmedWord );
+
+	// Check exceptions.
+	if ( exceptions.length > 0 ) {
+		forms.push( exceptions );
+		return unique( flattenDeep( forms ) );
```
[the remark this answers is not in the document]
Comment (manuelaugustin) [reply src/morphology/german/getForms.js]: Implemented as suggested; I just made sure to also add the original word to the exceptions array as a safeguard.

File: src/morphology/german/getForms.js (hunk 2 above)
Comment (manuelaugustin) [reply src/morphology/german/getForms.js]: ...completed!

**Comment by igorschoester** [comment]
Acceptance 👍 See https://github.com/Yoast/YoastSEO.js-premium-configuration/pull/7 for some more info.

**Pull request merged**